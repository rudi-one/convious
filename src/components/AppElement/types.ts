import {AppsCollection} from 'store/apps/types';

interface AppProps {
  app: AppsCollection;
}

export default AppProps;
