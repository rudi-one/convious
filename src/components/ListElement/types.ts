import {AppCollection} from 'store/apps/types';
import {MusicCollection} from 'store/music/types';

interface ElementProps {
  imageUrl: string;
  item: AppCollection | MusicCollection;
  showArtist?: boolean;
  navigate: any;
}

export default ElementProps;
