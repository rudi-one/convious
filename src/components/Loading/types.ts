export interface LoadingProps {
  mode?: string;
}
