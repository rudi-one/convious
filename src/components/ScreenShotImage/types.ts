export interface ScreenShotProp {
  url: string | undefined;
}
