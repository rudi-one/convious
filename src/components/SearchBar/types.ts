export interface SearchBarProps {
  onSearch: (text: string) => void;
}
