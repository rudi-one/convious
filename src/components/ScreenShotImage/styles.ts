import {StyleSheet} from 'react-native';
import {horizontalScale, verticalScale} from 'src/utils/helpers';

const styles = StyleSheet.create({
  screenshot: {
    width: horizontalScale(200),
    height: verticalScale(320),
  },
  screenshotContainer: {
    width: horizontalScale(205),
    height: verticalScale(320),
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
