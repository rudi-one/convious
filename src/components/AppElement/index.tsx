import {useNavigation} from '@react-navigation/native';
import {NavigationProp, RouteNames} from 'navigation/types';
import React, {memo} from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import AppProps from 'src/components/AppElement/types';
import styles from './styles';

const AppElementComponent = ({app}: AppProps) => {
  const navigation = useNavigation<NavigationProp>();

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() => {
        navigation.navigate(RouteNames.app, {app: app});
      }}>
      <View style={styles.imageWrapper}>
        <Image
          source={{
            uri: app.artworkUrl512,
          }}
          style={styles.image}
        />
      </View>
      <View style={styles.information}>
        <Text style={styles.genre}>{app.primaryGenreName}</Text>
        <Text style={styles.title}>{app.trackName}</Text>
      </View>
    </TouchableOpacity>
  );
};

export const AppElement = memo(AppElementComponent);
