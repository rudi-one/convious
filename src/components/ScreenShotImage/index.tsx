import React, {memo} from 'react';
import {Image, View} from 'react-native';
import styles from './styles';
import {ScreenShotProp} from './types';

const ScreenShotImageComponent = ({item}: {item: ScreenShotProp}) => {
  return (
    <View style={styles.screenshotContainer}>
      <Image
        source={{
          uri: item.url,
        }}
        style={styles.screenshot}
      />
    </View>
  );
};

export const ScreenShotImage = memo(ScreenShotImageComponent);
