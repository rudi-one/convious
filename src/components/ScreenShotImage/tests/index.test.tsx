import {render} from '@testing-library/react-native';
import React from 'react';
import {ScreenShotImage} from 'src/components/ScreenShotImage';
import {ScreenShotProp} from 'src/components/ScreenShotImage/types';
import {mockAppCollection1} from 'src/mocks/mockApps';

describe('<ScreenShotImage /> tests', () => {
  it('should match ScreenShotImage component snapshot', () => {
    const screenShot: ScreenShotProp = {
      url: mockAppCollection1.screenshotUrls[0],
    };
    const component = <ScreenShotImage item={screenShot} />;

    expect(render(component)).toMatchSnapshot();
  });
});
