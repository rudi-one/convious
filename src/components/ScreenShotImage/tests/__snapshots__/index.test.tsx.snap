// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenShotImage /> tests should match ScreenShotImage component snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 628,
      "justifyContent": "center",
      "width": 439.5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://is2-ssl.mzstatic.com/image/thumb/Purple126/v4/7d/94/dd/7d94ddff-2276-3fc2-65d9-a0e74fc63ce6/a3bfbfa4-3fd7-4391-8bf3-ae9e974acbed_en-US_iPhone_8_Plus_01_trips.png/392x696bb.png",
      }
    }
    style={
      Object {
        "height": 628,
        "width": 428.5,
      }
    }
  />
</View>
`;
